controller/update_test.go

```diff
@@ -207,6 +212,38 @@ func TestIsGroupRemovalAllowed(t *testing.T) {
 	}
 }
 
+// TestUpdateCurrentSliceIDs tests the updateCurrentSliceIDs method.
+func TestUpdateCurrentSliceIDs(t *testing.T) {
+	tests := []struct {
+		currentSliceIDs  []string
+		modifiedSliceIDs []string
+		newSliceIDs      []string
+		output           []string
+	}{
+		{
+			currentSliceIDs:  []string{"lol", "kek"},
+			modifiedSliceIDs: []string{"lol"},
+			newSliceIDs:      []string{"02b"},
+			output:           []string{"kek", "02b"},
+		},
```

Comment:
IMO some combinations are missing. Lets add some tests having more slice IDs and even none slice IDs. That way we can make sure more "complicated" cases are working and the code does not panic when lists are empty.